kyaml/fn/framework/processors_test.go

```diff
@@ -525,3 +527,76 @@ spec:
 		})
 	}
 }
+
+func TestTemplateProcessor_AdditionalSchemas(t *testing.T) {
+	p := framework.TemplateProcessor{
+		AdditionalSchemas: func() ([]*spec.Definitions, error) {
+			// This adds the same thing twice, just to exercise both the ...FromDir and the ...FromFile helpers
+			c1, err := framework.SchemaDefinitionsFromDir("/fn/framework/testdata/template-processor/schemas")()
+			if err != nil {
+				return nil, errors.WrapPrefixf(err, "schema from dir")
+			}
+			c2, err := framework.SchemaDefinitionsFromFile("/fn/framework/testdata/template-processor/schemas/foo.json")()
+			if err != nil {
+				return nil, errors.WrapPrefixf(err, "schema from file")
+			}
+			return append(c1, c2...), nil
+		},
+		ResourceTemplates: []framework.ResourceTemplate{{
+			Templates: framework.TemplatesFromFile("/fn/framework/testdata/template-processor/templates/custom-resource/foo.yaml"),
+		}},
+		PatchTemplates: []framework.PatchTemplate{
+			&framework.ResourcePatchTemplate{
+				Templates: framework.TemplatesFromFile("/fn/framework/testdata/template-processor/patches/custom-resource/patch.template.yaml")},
+		},
+	}
+	out := new(bytes.Buffer)
+
+	rw := &kio.ByteReadWriter{Reader: bytes.NewBufferString(`
+apiVersion: config.kubernetes.io/v1alpha1
+kind: ResourceList
+items:
+- apiVersion: example.com/v1
+  kind: Foo
+  metadata:
+    name: source
+  spec:
+    targets:
+    - app: C
+      size: medium
+`),
+		Writer: out}
+	defer openapi.ResetOpenAPI()
```

Comment:
Is this the right way to clean up? 🤔 in any case, I should probably do the cleanup right in TemplateProcessor#Filter, not here.

Reply:
yes, ResetOpenAPI() will remove all existing openapi definitions from the global schema. We should do that when TemplateProcessor is finished using them